src/utils/log/logFactory.ts

```diff
@@ -0,0 +1,35 @@
+import { TFunction } from 'i18next';
+import { generateImportLogEntries } from './importEntryGenerator';
+import { generateRegistrationLogEntries } from './registrationEntryGenerator';
+import { generateTicketLogEntries } from './ticketEntryGenerator';
+import { getArchivedFiles } from '../registration-helpers';
+import { Log, LogEntry } from '../../types/log.types';
+import { Ticket } from '../../types/publication_types/ticket.types';
+import { Registration } from '../../types/registration.types';
+
+export function generateLog(registration: Registration, tickets: Ticket[], t: TFunction): Log {
+  const importLogEntries = generateImportLogEntries(registration.importDetails ?? [], t);
+  const registrationLogEntries = generateRegistrationLogEntries(registration, tickets, t);
+  const ticketLogEntries = generateTicketLogEntries(tickets, registration, t);
+
+  const entries = [...importLogEntries, ...registrationLogEntries, ...ticketLogEntries];
+
+  return {
+    entries: entries.sort(sortByModifiedDateAndThenMetadataPublished),
+    metadataUpdated: registration.modifiedDate,
+    numberOfArchivedFiles: getArchivedFiles(registration.associatedArtifacts, tickets).length,
+  };
+}
+
+const sortByModifiedDateAndThenMetadataPublished = (a: LogEntry, b: LogEntry) => {
```

Comment:
Gjenbruker ticket for "metadata publisert" logginnslag, det vil derfor være tickets med samme timestamp, og da vil vi vise metadata først.